packages/legend-extension-dsl-data-space/src/stores/query/DataSpaceTemplateQueryPromotionReviewerStore.ts

```diff
@@ -0,0 +1,666 @@
+            this.currentQuery.groupId,
+            this.currentQuery.artifactId,
+          )) as PlainObject<StoreProjectData>,
+        );
+        const projectData = (yield Promise.all([
+          this.depotServerClient.getVersionEntities(
+            this.currentQuery.groupId,
+            this.currentQuery.artifactId,
+            this.currentQuery.versionId,
+          ),
+          this.sdlcServerClient.getConfiguration(
+            this.currentQueryProject.projectId,
+            undefined,
+          ),
+        ])) as [Entity[], PlainObject<ProjectConfiguration>];
+
+        const [currentProjectEntities, currentProjectConfiguration] = [
+          projectData[0],
+          ProjectConfiguration.serialization.fromJson(projectData[1]),
+        ];
+        this.currentProjectConfiguration = currentProjectConfiguration;
+        const dependencyEntities = (
+          (yield this.depotServerClient.collectDependencyEntities(
+            (
+              [
+                ...currentProjectConfiguration.projectDependencies,
+              ] as ProjectDependency[]
+            )
+              .map(projectDependencyToProjectCoordinates)
+              .map((p) => ProjectDependencyCoordinates.serialization.toJson(p)),
+            true,
+            true,
+          )) as PlainObject<ProjectVersionEntities>[]
+        )
+          .map((p) => ProjectVersionEntities.serialization.fromJson(p))
+          .flatMap((info) => info.entities);
+        this.dependencyEntities = dependencyEntities;
+        yield this.graphManagerState.graphManager.buildGraph(
+          this.graphManagerState.graph,
+          [...dependencyEntities, ...currentProjectEntities],
+          this.graphManagerState.graphBuildState,
+        );
+
+        this.dataSpace = getDataSpace(
+          dataSpacePath,
+          this.graphManagerState.graph,
+        );
+        this.initState.pass();
+      }
+    } catch (error) {
+      assertErrorThrown(error);
+      this.applicationStore.logService.error(
+        LogEvent.create(LEGEND_STUDIO_APP_EVENT.GENERIC_FAILURE),
+        error,
+      );
+      this.applicationStore.alertService.setBlockingAlert({
+        message: `Can't initialize template query promotion reviewer store`,
+      });
+      this.initState.fail();
+    }
+  }
+
+  *loadQuery(query: LightQuery): GeneratorFn<void> {
+    this.currentQuery = query;
+    this.templateQueryTitle = query.name;
+
+    try {
+      this.loadQueryState.inProgress();
+      this.currentQueryInfo =
+        (yield this.graphManagerState.graphManager.getQueryInfo(
+          query.id,
+        )) as QueryInfo;
+      this.currentQueryProject = StoreProjectData.serialization.fromJson(
+        (yield this.depotServerClient.getProject(
+          this.currentQuery.groupId,
+          this.currentQuery.artifactId,
+        )) as PlainObject<StoreProjectData>,
+      );
+      this.setWorkspaceName(`${DEFAULT_WORKSPACE_NAME_PREFIX}-${query.id}`);
```

Comment:
lets test this workspace name, sometimes gitlab build fails with workspace names like these